### DISCUSSION TITLE
Expected behavior for `jsonpath` in `when` clause if path is missing

### DISCUSSION THREAD

**Discussion by apiwoni** [argoproj/argo-workflows#12546] [Q&A] [answered]
What is expected behavior in `when` clause with
```
{{=jsonpath(tasks.TaskName.outputs.parameters.parameterName,'$.propertyName)}} != true
```
condition if the path does not exist, either no outputs or missing property?

It seems we get `Invalid 'when' expression` error.
Is there a way to test path existence if that's the case?
Is there a way to test for property existence at a given path?

**Comment by apiwoni** [comment, marked as answer]
I'm answering my question here but additional ideas and notes are welcome.

So yes, `invalid 'when' expression` also happens when actual path is missing for `jsonpath`.

Here's one way to handle it:
```
tasks.TaskName.outputs != nil && 
tasks.TaskName.outputs.parameters.parameterName != nil &&
tasks.TaskName.outputs.parameters.parameterName contains 'propertyName' &&
jsonpath(tasks.TaskName.outputs.parameters.parameterName, '$.propertyName') != 'true'
```

**Reply by agilgur5** [reply]
You can also use [optional chaining](https://expr-lang.org/docs/language-definition#optional-chaining) with `?.` to simplify the logic a bit
